acct-user/spamd/spamd-0.ebuild

```diff
@@ -0,0 +1,16 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit acct-user
+
+DESCRIPTION="User for the SpamAssassin daemon"
+
+ACCT_USER_ID=337
+ACCT_USER_GROUPS=( spamd )
+# The spamd daemon runs as this user. Use a real home directory so
+# that it can hold SA configuration.
+ACCT_USER_HOME="/home/spamd"
```

Comment:
Ebuilds aren't allowed to install to `/home` (we have this exact same problem with the amavis user, whose SA configuration is stored in `$HOME`). At least in this case, though, something like `/var/lib/spamd` should work? You can probably tighten the permissions on it, too: http://michael.orlitzky.com/articles/gentoo_glep81_user_package_guidelines.xhtml#home-directory-permissions

Reply:
I know I'm going to regret asking this ( ;-) ), but do we/should we migrate existing `/home/spamd` to `/var/lib/spamd`?  (I went with `/home/spamd`/ because that's what was already being used.)

Reply:
I think I got away with it in the past because the call to useradd in user.eclass sidesteps the package manager. As a result, there aren't any files owned by spamassassin installed to `/home/spamd`. But now with the GLEP81 eclass, we'll wind up calling keepdir on `/home/spamd` and I think that will generate a QA warning. (It's possible that the QA warning is obsolete in light of GLEP81, but that's a whole different argument.)

Migration should be pretty easy -- stop the daemon, move the directory, start the daemon. I had to move the directory in net-analyzer/nagios-core too, and settled on just showing a warning to people who are upgrading.

As for EAPI=7, I think we can do it, it will just take a little bit of thought. Everything that needs to _run_ during the build step has to go in ` BDEPEND`. I think that also includes the dependencies that the Makefile will check for, since the Makefile will run on the build host, and otherwise it will fail with "missing dependencies." The perl team might be able to help, but if none of us are cross-compiling, we may just have to settle for a best guess.

Reply:
I asked in #gentoo-perl and got some pointers:

> **kent\n:** the redux I think is: use DEPEND for anything with binary linking, eg: If you have a perl library that links (through XS) to libxml, the ebuild for that library should DEPEND on libxml, but otherwise, BDEPEND/RDEPEND everything
>
> **kent\n:** and the majority of perl deps are RDEPEND, and BDEPEND="${RDEPEND} ... <build deps> test? ( <test deps> )"

Additionally kent\n clarified that DEPEND is only for if the current package (ie: spamassassin) has a binary linking requirement on it's immediate dependencies.  Eg: spamassassin doesn't need to worry that dev-perl/Geo-IP happens to have a binary dependency on dev-perl/Geo-IP -- spamassassin just depends on the perl-code side of it, so dev-perl/Geo-IP goes into RDEPEND (and probably BDEPEND when test is set).

Reply:
Yeah, it looks to me like openssl/libressl are the only things that should stay in `DEPEND`, and the rest of `DEPEND` should move to `BDEPEND`. The test dependencies may be a little bit wrong since no one has been able to give me a straight answer on bug 693916 (and since I don't cross-compile myself), but that should get it "good enough."